android/src/main/java/io/jari/fingerprint/FingerprintModule.java

```diff
@@ -14,171 +13,125 @@
+    public final void hasEnrolledFingerprints(final Promise promise) {
+        promise.resolve(fpm.hasEnrolledFingerprints());
     }
 
     @ReactMethod
-    public void isHardwareDetected(Promise promise) {
-        try {
-            promise.resolve(fingerprintManager.isHardwareDetected());
-        } catch (SecurityException secEx) {
-            Exception exception = new Exception("App does not have the proper permissions. (did you add USE_FINGERPRINT to your manifest?)\nMore info see https://github.com/jariz/react-native-fingerprint-android");
-            exception.initCause(secEx);
-            promise.reject(exception);
-        } catch (Exception ex) {
-            promise.reject(ex);
-        }
+    @SuppressWarnings("unused")
+    public final void isHardwareDetected(final Promise promise) {
+        promise.resolve(fpm.isHardwareDetected());
     }
 
     @ReactMethod
-    public void authenticate(Promise promise) {
+    @SuppressWarnings("unused")
+    public final void authenticate(final Promise promise) {
         try {
-            isCancelled = false;
-            cancellationSignal = new CancellationSignal();
-            fingerprintManager.authenticate(null, cancellationSignal, 0, new AuthenticationCallback(promise), null);
-        } catch (SecurityException secEx) {
-            Exception exception = new Exception("App does not have the proper permissions. (did you add USE_FINGERPRINT to your manifest?)\nMore info see https://github.com/jariz/react-native-fingerprint-android");
-            exception.initCause(secEx);
-            promise.reject(exception);
-        } catch (Exception ex) {
-            promise.reject(ex);
+            IS_CANCELED = false;
+            CANCELLATION_SIGNAL = new CancellationSignal();
+            fpm.authenticate(
+                    null,
+                    0,
+                    CANCELLATION_SIGNAL,
+                    new FingerprintManagerCompat.AuthenticationCallback() {
+                        @Override
+                        public final void onAuthenticationError(
+                                final int errorCode,
+                                final CharSequence errorString
+                        ) {
+                            super.onAuthenticationError(errorCode, errorString);
+                            IS_CANCELED = CANCELLATION_SIGNAL.isCanceled();
+                            promise.reject(Integer.toString(errorCode), errorString.toString());
+                        }
+
+                        @Override
+                        public final void onAuthenticationHelp(
+                                final int helpCode,
+                                final CharSequence helpString
+                        ) {
+                            super.onAuthenticationHelp(helpCode, helpString);
+                            final WritableNativeMap writableNativeMap = new WritableNativeMap();
+                            writableNativeMap.putInt("code", helpCode);
```

Comment:
Passing this code is pretty useless now, because we have no way of telling what it stands for in javascript anymore.